iroh-resolver/src/resolver.rs

```diff
@@ -581,6 +617,42 @@ impl<T: ContentLoader> Resolver<T> {
         }
     }
 
+    /// Resolve a path recursively and yield the raw bytes plus metadata.
+    #[tracing::instrument(skip(self))]
+    pub fn resolve_recursive_raw(&self, root: Path) -> impl Stream<Item = Result<OutRaw>> {
```

Comment:
For security reasons (very easy to DOS a gateway with this currently), could you add a recursion limit please, maybe just as a parameter to the function and then passing a reasonable constant from the gateway?

Reply:
I added a simple recursion limit in the latest commit.